tests/inst/src/repobin.rs

```diff
@@ -0,0 +1,135 @@
+        test_nofifo,
+        test_mtime,
+        test_extensions,
+        test_pull_basicauth
+    )
+}
+
+fn test_nofifo(tmp_dir: &Path) -> Result<()> {
+    sh_execute!(
+        r"cd {tmp_dir}
+    ostree --repo=repo init --mode=archive
+    mkdir tmproot
+    mkfifo tmproot/afile
+",
+        tmp_dir = tmp_dir.to_str()
+    )?;
+    cmd_fails_with(
+        sh_command!(
+            r#"cd {tmp_dir}
+ls -al
+ostree --repo=repo commit -b fifotest -s "commit fifo" --tree=dir=./tmproot"#,
+            tmp_dir = tmp_dir.to_str()
+        )
+        .unwrap(),
+        "Not a regular file or symlink",
+    )?;
+    Ok(())
+}
+
+fn test_mtime(tmp_dir: &Path) -> Result<()> {
+    sh_execute!(
+        r"cd {tmp_dir}
+    ostree --repo=repo init --mode=archive
+    mkdir tmproot
+    echo afile > tmproot/afile
+    ostree --repo=repo commit -b test --tree=dir=tmproot >/dev/null
+",
+        tmp_dir = tmp_dir.to_str()
+    )?;
+    let ts = tmp_dir.join("repo").metadata()?.modified().unwrap();
+    sh_execute!(
+        r#"cd {tmp_dir}
+    ostree --repo=repo commit -b test -s "bump mtime" --tree=dir=tmproot >/dev/null"#,
+        tmp_dir = tmp_dir.to_str()
+    )?;
+    assert_ne!(ts, tmp_dir.join("repo").metadata()?.modified().unwrap());
+    Ok(())
+}
+
+fn test_extensions(tmp_dir: &Path) -> Result<()> {
+    sh_execute!(
+        r"ostree --repo={tmp_dir}/repo init --mode=bare",
+        tmp_dir = tmp_dir.to_str()
+    )?;
+    assert!(tmp_dir.join("repo/extensions").exists());
+    Ok(())
+}
+
+async fn impl_test_pull_basicauth(tmp_dir: &Path) -> Result<()> {
+    let opts = TestHttpServerOpts {
+        basicauth: true,
+        ..Default::default()
+    };
+    let serverrepo = tmp_dir.join("server/repo");
+    std::fs::create_dir_all(&serverrepo)?;
+    let addr = http_server(&serverrepo, opts).await?;
+    let tmp_dir = tmp_dir.to_path_buf();
+    tokio::task::spawn_blocking(move || -> Result<()> {
+        let baseuri = http::Uri::from_maybe_shared(format!("http://{}/", addr).into_bytes())?;
+        let unauthuri =
+            http::Uri::from_maybe_shared(format!("http://unknown:badpw@{}/", addr).into_bytes())?;
+        let authuri = http::Uri::from_maybe_shared(
+            format!("http://{}@{}/", TEST_HTTP_BASIC_AUTH, addr).into_bytes(),
+        )?;
+        let osroot = tmp_dir.join("osroot");
+        mkroot(&osroot)?;
+        sh_execute!(
+            r#"cd {tmp_dir}
+        ostree --repo={serverrepo} init --mode=archive
+        ostree --repo={serverrepo} commit -b os --tree=dir={osroot} >/dev/null
+        mkdir client
+        cd client
+        ostree --repo=repo init --mode=archive
+        ostree --repo=repo remote add --set=gpg-verify=false origin-unauth {baseuri}
+        ostree --repo=repo remote add --set=gpg-verify=false origin-badauth {unauthuri}
+        ostree --repo=repo remote add --set=gpg-verify=false origin-goodauth {authuri}
+        "#,
+            tmp_dir = tmp_dir.to_str(),
```

Comment:
I'm probably going to do another commandspec PR soon to allow just saying `tmp_dir`.  Or maybe what we should really do is just fork each of these tests as a subprocess in their own tmpdir.